pkg/distributor/distributor_test.go

```diff
@@ -2461,35 +2485,65 @@ func TestDistributor_LabelNames(t *testing.T) {
 	}
 
 	for testName, testData := range tests {
+		testData := testData
+
 		t.Run(testName, func(t *testing.T) {
-			now := model.Now()
+			for _, ingestStorageEnabled := range []bool{false, true} {
```

Comment:
you can add t.Parallel here as well so that the different `testName`s are ran in parallel. Currently only with and without ingest storage are ran in parallel, but different testName/testData pairs run sequentially. This applies to a few other tests too

Reply:
Done in https://github.com/grafana/mimir/pull/7393/commits/cff83c8f09893bfb14d253bb3c005be43444e554